### PR TITLE
fix Mental Health, Elderly and Palliative Care screening modules [MHWC-173, MHWC-175, MHWC-176, MHWC-178, MHWC-181]

### PR DESCRIPTION
**Pull request by abhilashkadu-tekdi**
## 📋 Description

JIRA ID: MHWC-173, MHWC-175, MHWC-176, MHWC-178, MHWC-181

fix Mental Health, Elderly and Palliative Care screening modules

---

## ✅ Type of Change

- [x] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [ ] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Summary by CodeRabbit

* **New Features**
  * ADL (Activities of Daily Living) block in elderly screening with automatic scoring and functional status
  * Basic symptom assessment & basic management section for palliative/pain screening
  * PHQ‑9 alert flow that shows non-cancelable alert dialogs for elevated scores

* **UI Improvements**
  * Mental health form: title sizing, submit button now shows "Next"
  * Updated edit icon styling in patient list

* **Localization**
  * Many form labels, options and alerts localized; PHQ‑9 alerts highlighted in red

### CODE REVIEW

**Comment by abhilashkadu-tekdi** [comment]
@coderabbitai full review

File: app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt [hunk 1]
```diff
@@ -906,88 +1209,124 @@ class MentalHealthScreeningDataset(
...
+
+            // Yes No No No No -> Low
+            isThoughtsYes && isPreviousNo && isIntentNo && isMeansNo && isImmediateNo -> low
+
+            // No No No No No -> Low
+            isThoughtsNo && isPreviousNo && isIntentNo && isMeansNo && isImmediateNo -> low
+
+            // Unlisted combinations
+            else -> low
         }
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt:1376]: This flow is correct as per brd

File: app/src/main/java/org/piramalswasthya/cho/configuration/ElderlyHealthAssessmentDataset.kt [hunk 2]
```diff
@@ -154,6 +237,12 @@ class ElderlyHealthAssessmentDataset(
         list.add(difficultyWalkingBalance)
         list.add(visualHearingDifficulty)
         list.add(functionalDecline)
+
+        if (functionalDecline.value == "Yes") {
+            list.addAll(getADLFields())
+            computeADLScore()
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/ElderlyHealthAssessmentDataset.kt]: Resolved

File: app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt [hunk 3]
```diff
@@ -1126,6 +1467,7 @@ class MentalHealthScreeningDataset(
             cache.followUpRequired?.let { if (it) yesNoOptions[0] else yesNoOptions[1] }
         mhFollowUpDate.value = cache.followUpDate
         mhImprovementNoted.value = cache.improvementNoted
+        mhAdherenceToAdvice.value = cache.caseClosureReason
         mhReferralEscalation.value =
             cache.referralEscalationRequired?.let { if (it) yesNoOptions[0] else yesNoOptions[1] }
         mhCaseClosureReason.value = cache.caseClosureReason
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt:1522]: Resolved

File: app/src/main/java/org/piramalswasthya/cho/database/room/InAppDb.kt [hunk 4]
```diff
@@ -1117,6 +1117,37 @@ abstract class InAppDb : RoomDatabase() {
...
+                safeAddColumn(database, "ELDERLY_HEALTH_ASSESSMENT", "total_score", "INTEGER")
+                safeAddColumn(database, "ELDERLY_HEALTH_ASSESSMENT", "functional_status", "TEXT")
+                safeAddColumn(database, "ELDERLY_HEALTH_ASSESSMENT", "functional_decline_flag", "INTEGER")
+            }
+        }
+        val MIGRATION_143_144 = object : Migration(143, 144) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                safeAddColumn(database, "MENTAL_HEALTH_SCREENING", "ed_reason", "TEXT")
+                safeAddColumn(database, "MENTAL_HEALTH_SCREENING", "ed_reason", "TEXT")
+            }
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/database/room/InAppDb.kt:1137]: Resolved.

File: app/src/main/java/org/piramalswasthya/cho/model/MentalHealthScreeningCache.kt [hunk 5]
```diff
@@ -195,6 +195,8 @@ data class MentalHealthScreeningCache(
     var edRecurrentEpisodeloss: Boolean? = null,
     @ColumnInfo(name = "ed_recurrent_jerky_movements")
     var edRecurrentJerkyMovements: Boolean? = null,
+    @ColumnInfo(name = "ed_confusion_ordrowsiness")
+    var edConfusionordrowsiness: Boolean? = null,
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/model/MentalHealthScreeningCache.kt:200]: Resolved.

File: app/src/main/java/org/piramalswasthya/cho/ui/commons/BaseAssessmentFormFragment.kt [hunk 6]
```diff
@@ -108,7 +108,13 @@ abstract class BaseAssessmentFormFragment<VM : BaseFormViewModel> : Fragment(),
 
         viewLifecycleOwner.lifecycleScope.launch {
             getFormFlow().collect { list ->
-                if (list.isNotEmpty()) adapter.submitList(list)
+                if (list.isNotEmpty()) {
+                    adapter.submitList(list.toMutableList()) {
+                        // Force an immediate rebind so newly shown/updated dynamic fields
+                        // are reflected without requiring manual scroll.
+                        inputFormRecyclerView.post { adapter.notifyDataSetChanged() }
+                    }
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/ui/commons/BaseAssessmentFormFragment.kt:115]: Resolved.

File: app/src/main/java/org/piramalswasthya/cho/ui/commons/fhir_visit_details/FragmentVisitDetail.kt [hunk 7]
```diff
@@ -390,7 +390,7 @@ class FragmentVisitDetail : Fragment(), NavigationAdapter,
                     requireContext(),
                     R.layout.dropdown_subcategory,
                     R.id.tv_dropdown_item_text,
-                    listOf(DropdownConst.oral)
+                    listOf(DropdownConst.oral,DropdownConst.mentalHealth)
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/ui/commons/fhir_visit_details/FragmentVisitDetail.kt:393]: Resolved.

File: app/src/main/java/org/piramalswasthya/cho/ui/mental_health/MentalHealthScreeningFormFragment.kt [hunk 8]
```diff
@@ -96,4 +104,24 @@ class MentalHealthScreeningFormFragment :
...
+                            .setPositiveButton(android.R.string.ok) { dialog, _ ->
+                                dialog.dismiss()
+                                viewModel.clearPhq9AlertMessage()
+                            }
+                            .setCancelable(false)
+                            .show()
+                    }
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/ui/mental_health/MentalHealthScreeningFormFragment.kt:130]: Resolved.

**Comment by abhilashkadu-tekdi** [comment]
@coderabbitai full review

File: app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt [hunk 9]
```diff
@@ -1,91 +1,106 @@
...
+    private val optionYes = context.getString(R.string.yes_option)
+    private val optionNo = context.getString(R.string.no_option)
 
     private val difficultyHearing = FormElement(
         id = 1,
         inputType = InputType.RADIO,
-        title = "Difficulty Hearing",
-        entries = arrayOf("Yes", "No"),
+        title = context.getString(R.string.ear_difficulty_hearing),
+        entries = arrayOf(optionYes, optionNo),
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt:26]: Our implementation is correct because we are already using the context from the class when it loads, so it will work.

File: app/src/main/java/org/piramalswasthya/cho/ui/mental_health/MentalHealthScreeningFormFragment.kt [hunk 10]
```diff
@@ -9,7 +9,12 @@ import androidx.fragment.app.viewModels
 import androidx.navigation.fragment.findNavController
 import androidx.recyclerview.widget.RecyclerView
 import dagger.hilt.android.AndroidEntryPoint
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.repeatOnLifecycle
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/ui/mental_health/MentalHealthScreeningFormFragment.kt:14]: This change is not required in the current scope .

File: app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt [hunk 11]
```diff
@@ -1,91 +1,106 @@
...
     private val whisperTestResponse = FormElement(
         id = 2,
         inputType = InputType.DROPDOWN,
-        title = "Whisper Test Response",
-        entries = arrayOf("Correct", "Incorrect"),
+        title = context.getString(R.string.ear_whisper_test_response),
+        entries = arrayOf(
+            context.getString(R.string.ear_whisper_correct),
+            context.getString(R.string.ear_whisper_incorrect)
+        ),
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt:38]: Our implementation is correct because we are already using the context from the class when it loads, so it will work.

File: app/src/main/res/values-kn-rIN/strings.xml [hunk 12]
```diff
@@ -574,4 +572,478 @@
...
+    </string-array>
+    
+    <string name="follow_up_date_title">ಮುಂದಿನ ಭೇಟಿ ದಿನಾಂಕ</string>
+    
+    <string name="case_status_title">ಪ್ರಕರಣದ ಸ್ಥಿತಿ</string>
+    <string-array name="case_status_options">
+        <item>ಚೇತರಿಸಿಕೊಂಡಿದ್ದಾರೆ</item>
+        <item>ಚಿಕಿತ್ಸೆಯಲ್ಲಿದ್ದಾರೆ</item>
+        <item>ಸಾವು</item>
+    </string-array>
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/res/values-kn-rIN/strings.xml:822]: The  array is already defined in arrays.xml within the base values directory.

File: app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt (hunk 9 above)
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt:26]: Resolved

File: app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt (hunk 11 above)
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/EarDiagnosisDataset.kt:38]: Resolved

File: app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt [hunk 13]
```diff
@@ -558,31 +675,220 @@ class MentalHealthScreeningDataset(
...
     ) {
+        this.genderID = genderID
+        this.age = age
         cache = savedRecord ?: MentalHealthScreeningCache(
             patientID = "",
             benVisitNo = null
         )
 
-        cache.isPostpartum = isPostpartumFromRmncha
+        cache.isPostpartum = if (genderID == 2 && age in 15..49) isPostpartumFromRmncha else null
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt:692]: Resolved

File: app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt [hunk 14]
```diff
@@ -558,31 +675,220 @@ class MentalHealthScreeningDataset(
...
+        return isYes(emotionalBehaviouralConcerns.value) ||
+                isYes(substanceUseConcerns.value) ||
+                isYes(selfHarmSuicideThoughts.value) ||
+                isYes(memoryLossConfusion.value) ||
+                isYes(seizuresFitsLoc.value) ||
+                isYes(isPostpartum.value) ||
+                cache.isPostpartum == true ||
+                shouldAutoReferFromPhq9() ||
+                shouldAutoReferFromHighSuicideRisk()
+    }
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt:720]: Resolved

File: app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt [hunk 15]
```diff
@@ -1271,22 +1669,22 @@ class MentalHealthScreeningDataset(
...
+                it.substanceTobaccoUse = isYes(substanceTobaccoUse.value)
+                it.substance_alcohol_loss = isYes(substance_alcohol_loss.value)
+                it.substanceAlcoholImpact = isYes(substanceAlcoholImpact.value)
+                it.substanceAlcoholWithdrawal = isYes(substanceAlcoholWithdrawal.value)
+                it.substanceAlcoholProblematic = isYes(substanceAlcoholProblematic.value)
                 it.substanceAlcoholClassification = substanceAlcoholClassification.value
                 it.substanceAlcoholSystemAction = substanceAlcoholSystemAction.value
                 it.substance_alcohol_frequency = substance_alcohol_frequency.value
-                it.briefInterventionGiven = briefInterventionGiven.value == "Yes"
+                it.briefInterventionGiven = isYes(briefInterventionGiven.value)
```
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/configuration/MentalHealthScreeningDataset.kt]: Resolved

File: app/src/main/java/org/piramalswasthya/cho/ui/mental_health/MentalHealthScreeningFormFragment.kt (hunk 10 above)
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/java/org/piramalswasthya/cho/ui/mental_health/MentalHealthScreeningFormFragment.kt:14]: Resolved

File: app/src/main/res/values-kn-rIN/strings.xml (hunk 12 above)
[the remark this answers is not in the document]
Comment (abhilashkadu-tekdi) [reply app/src/main/res/values-kn-rIN/strings.xml:822]: Resolved

**Review by DeepakKumar471 — APPROVED** [review]
